### PR TITLE
Add ansible 2.9.7

### PR DESCRIPTION
**Pull request by alex1a**
Add **ansible** v2.9.7 package

### CODE REVIEW

**Comment by alex1a** [comment]
ci is failing because symlinks on windows, i can't exctract source file correctly

**Comment by alex1a** [comment]
ci is failing because symlinks on windows

**Comment by Biswa96** [comment]
I would like to try to build this.

* Do you have a build instruction for Windows?
* Is it possible to make this package in mingw mode?

**Comment by alex1a** [comment]
Related to #1216

File: ansible/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+optdepends=('sshpass: for ssh connections with password')
+makedepends=('tar' 'gcc' 'python3' 'python3-pip' 'libffi' 'libffi-devel' 'libcrypt' 'libcrypt-devel' 'openssl' 'openssl-devel')
+backup=('etc/ansible/ansible.cfg')
+source=("https://releases.ansible.com/ansible/ansible-${pkgver}.tar.gz")
+sha512sums=('ce029441bcafdc5b44c9fda69f183d4defea84ead5628164caf87306cb97efec68c11b2cce728e90f28290640c320549486a6b4e823710f638d1d2e7c35675a4')
+noextract=("${pkgname}-${pkgver}.tar.gz")
+
+prepare() {
+  [[ -d ${pkgname}-${pkgver} ]] && rm -rf ${pkgname}-${pkgver}
+  export MSYS=winsymlinks:lnk
```
Comment (alex1a) [inline ansible/PKGBUILD:21]: Related to #1216. This work for me, any better solution???

**Comment by alex1a** [comment]
This is how i reproduce in my msys shell, any hints why ci build is failing?

```
(cd ansible; rm -rf pkg/ src/; rm *.pkg.tar.xz; pacman -Rs --noconfirm ansible)
(cd python-wcwidth; rm -rf pkg/ src/; rm *.pkg.tar.xz; pacman -Rs --noconfirm python-wcwidth)
(cd python-jinja; rm -rf pkg/ src/; rm *.pkg.tar.xz; pacman -Rs --noconfirm python-jinja)
(cd python-yaml; rm -rf pkg/ src/; rm *.pkg.tar.xz; pacman -Rs --noconfirm python-yaml)
(cd cython; rm -rf pkg/ src/; rm *.pkg.tar.xz; pacman -Rs --noconfirm cython)

(cd cython; makepkg -si --noconfirm)
(cd python-yaml; makepkg -si --noconfirm)
(cd python-jinja; makepkg -si --noconfirm)
(cd python-wcwidth; makepkg -si --noconfirm)
(cd ansible; makepkg -si --noconfirm)
```

**Comment by lazka** [comment]
The CI errors have been fixed in master (rebasing should help). Sorry :/

**Comment by alex1a** [comment]
> 
> 
> The CI errors have been fixed in master (rebasing should help). Sorry :/

No problem, so far I've been using ansible without problems. And now ci is fixed, thank you

File: ansible/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Maintainer: Alexandre Ferreira < alex dot jorge dot m at gmail dot com >
+
+pkgname=ansible
+pkgver=2.9.7
+pkgrel=1
+pkgdesc='Radically simple IT automation platform'
+arch=('i686' 'x86_64')
+url='https://www.ansible.com'
+license=('GPL3')
+depends=('python3' 'python-yaml' 'python-jinja')
```
Comment (Alexpux) [inline ansible/PKGBUILD:10]: Rename to python. We renamed all python3 packages already

File: ansible/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+provides=('python-ansible')
+optdepends=('sshpass: for ssh connections with password')
+makedepends=('tar' 'gcc' 'python3' 'python3-pip' 'libffi' 'libffi-devel' 'libcrypt' 'libcrypt-devel' 'openssl' 'openssl-devel')
```
Comment (Alexpux) [inline ansible/PKGBUILD:13]: also rename all python3 parts. In makedepends you need to leave only devel parts of packages. Others you need to check and move to depends

File: python-jinja/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# Maintainer: Alexandre Ferreira < alex dot jorge dot m at gmail dot com >
+
+_realname=jinja
+pkgbase=python-${_realname}
+pkgname=("python-${_realname}")
+provides=("python3-${_realname}")
```
Comment (Alexpux) [inline python-jinja/PKGBUILD:6]: We previously dont have this package so remove provides/conflicts/replaces

File: python-wcwidth/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Maintainer: Alexandre Ferreira < alex dot jorge dot m at gmail dot com >
+
+_realname=wcwidth
+pkgbase=python-${_realname}
+pkgname=("python-${_realname}")
+provides=("python3-${_realname}")
```
Comment (Alexpux) [inline python-wcwidth/PKGBUILD:6]: We previously dont have this package so remove provides/conflicts/replaces

File: python-yaml/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,51 @@
+# Maintainer: Alexandre Ferreira < alex dot jorge dot m at gmail dot com >
+
+_pyname=PyYAML
+_realname=yaml
+pkgbase=python-${_realname}
+pkgname=("python-${_realname}")
+provides=("python3-${_realname}")
```
Comment (Alexpux) [inline python-yaml/PKGBUILD:7]: We previously dont have this package so remove provides/conflicts/replaces

**Comment by Alexpux** [comment]
@alex1a ping

File: ansible/PKGBUILD (hunk 1 above)
Comment (alex1a) [reply ansible/PKGBUILD:21]: @Alexpux exporting the MSYS variable will not affect the rest of the environment?

File: python-jinja/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+build() {
+  msg "Python build for ${CARCH}"
+  cd "${srcdir}/python-build-${CARCH}"
+  /usr/bin/python setup.py build
+}
+
+package() {
+  cd "${srcdir}/python-build-${CARCH}"
+  MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
```
Comment (Alexpux) [inline python-jinja/PKGBUILD:31]: For MSYS packages this doesnt need, only for MINGW packages

File: python-wcwidth/PKGBUILD [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+#check() {
+#    cd ${srcdir}/python-build-${CARCH}
+#    tox
+#}
+
+package() {
+  cd "${srcdir}/python-build-${CARCH}"
+  MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
```
Comment (Alexpux) [inline python-wcwidth/PKGBUILD:36]: The same

File: python-yaml/PKGBUILD [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+check() {
+  msg "Python test for ${CARCH}"
+  cd "${srcdir}/python-build-${CARCH}"
+  /usr/bin/python setup.py test
+}
+
+package() {
+  cd "${srcdir}/python-build-${CARCH}"
+  MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
```
Comment (Alexpux) [inline python-yaml/PKGBUILD:42]: No need

File: cython/PKGBUILD [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+  DISTUTILS_DEBUG=1 \
+  /usr/bin/python setup.py build
+}
+
+package() {
+  cd "${srcdir}/python-build-${CARCH}"
+  MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
+  DISTUTILS_DEBUG=1 \
+  /usr/bin/python setup.py install \
+                              --root="${pkgdir}" --optimize=1 --skip-build
```
Comment (Alexpux) [inline cython/PKGBUILD:35]: Write in one line

File: cython/PKGBUILD [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+build() {
+  cd "${srcdir}/python-build-${CARCH}"
+  DISTUTILS_DEBUG=1 \
+  /usr/bin/python setup.py build
+}
+
+package() {
+  cd "${srcdir}/python-build-${CARCH}"
+  MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
```
Comment (Alexpux) [inline cython/PKGBUILD:32]: No need

File: ansible/PKGBUILD (hunk 1 above)
Comment (Alexpux) [reply ansible/PKGBUILD:21]: @dont do that. Remove this export

File: cython/PKGBUILD [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+depends=('python-setuptools')
+makedepends=('gcc' 'libcrypt-devel')
+source=("${_realname}-${pkgver}.tar.gz"::"https://github.com/cython/cython/archive/${pkgver}.tar.gz")
+sha256sums=('5ab5c1e92fe334bcd3cde3b0c34dd96d1b007d64e7fa9fdbf53bb368c5d27e22')
+
+prepare() {
+  cd "${srcdir}"
+  rm -rf python-build-${CARCH}| true
+  cp -r "${_realname}-${pkgver}" "python-build-${CARCH}"
+  sed -e "s|\/usr\/bin\/python|\/usr\/bin\/python|" -i python-build-${CARCH}/setup.py
```
Comment (Alexpux) [inline cython/PKGBUILD:21]: What do you do here?

File: python-jinja/PKGBUILD [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+    /usr/bin/python setup.py install --root="${pkgdir}" -O1
+
+  install -Dm644 LICENSE.rst "${pkgdir}/share/licenses/python-${_realname}/LICENSE.rst"
```
Comment (Alexpux) [inline python-jinja/PKGBUILD:34]: You forgot /usr

File: python-wcwidth/PKGBUILD [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+  /usr/bin/python setup.py install --root="${pkgdir}" --optimize=1 --skip-build
+
+  install -Dm644 LICENSE.txt "${pkgdir}/share/licenses/python-${_realname}/LICENSE"
```
Comment (Alexpux) [inline python-wcwidth/PKGBUILD:39]: You forgot /usr

File: python-yaml/PKGBUILD [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+  /usr/bin/python setup.py --with-libyaml install \
+    --root="${pkgdir}" --optimize=1 --skip-build
+
+  install -Dm644 LICENSE "${pkgdir}/share/licenses/python-${_realname}/COPYING"
```
Comment (Alexpux) [inline python-yaml/PKGBUILD:46]: You forogt /usr

File: python-yaml/PKGBUILD [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+  install -Dm644 CHANGES README -t "${pkgdir}/share/doc/python-${_realname}"
```
Comment (Alexpux) [inline python-yaml/PKGBUILD:47]: the same

**Comment by Alexpux** [comment]
@alex1a fix other comments

File: ansible/PKGBUILD (hunk 1 above)
Comment (alex1a) [reply ansible/PKGBUILD:21]: @Alexpux removing `export MSYS=winsymlinks:lnk` lead to this errors extracting source, the better solution is to tar two times as mentioned here #1216?

```
tar: ansible-2.9.7/lib/ansible/modules/web_infrastructure/_jenkins_job_facts.py: Cannot create symlink to ‘jenkins_job_info.py’: No such file or directory
tar: ansible-2.9.7/lib/ansible/plugins/callback/osx_say.py: Cannot create symlink to ‘say.py’: No such file or directory
tar: ansible-2.9.7/lib/ansible/plugins/doc_fragments/ovirt_facts.py: Cannot create symlink to ‘ovirt_info.py’: No such file or directory
tar: ansible-2.9.7/lib/ansible/plugins/lookup/_openshift.py: Cannot create symlink to ‘k8s.py’: No such file or directory
tar: ansible-2.9.7/test/integration/targets/supervisorctl/tasks/install_Darwin.yml: Cannot create symlink to ‘install_pip.yml’: No such file or directory
tar: ansible-2.9.7/test/integration/targets/supervisorctl/tasks/install_FreeBSD.yml: Cannot create symlink to ‘install_pip.yml’: No such file or directory
tar: ansible-2.9.7/test/integration/targets/supervisorctl/tasks/install_RedHat.yml: Cannot create symlink to ‘install_pip.yml’: No such file or directory
tar: ansible-2.9.7/test/integration/targets/supervisorctl/tasks/install_Suse.yml: Cannot create symlink to ‘install_pip.yml’: No such file or directory
tar: ansible-2.9.7/test/integration/targets/supervisorctl/tasks/uninstall_Darwin.yml: Cannot create symlink to ‘uninstall_pip.yml’: No such file or directory
tar: ansible-2.9.7/test/integration/targets/supervisorctl/tasks/uninstall_FreeBSD.yml: Cannot create symlink to ‘uninstall_pip.yml’: No such file or directory
tar: ansible-2.9.7/test/integration/targets/supervisorctl/tasks/uninstall_RedHat.yml: Cannot create symlink to ‘uninstall_pip.yml’: No such file or directory
tar: ansible-2.9.7/test/integration/targets/supervisorctl/tasks/uninstall_Suse.yml: Cannot create symlink to ‘uninstall_pip.yml’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/ansible: Cannot create symlink to ‘python.py’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/ansible-config: Cannot create symlink to ‘python.py’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/ansible-connection: Cannot create symlink to ‘python.py’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/ansible-console: Cannot create symlink to ‘python.py’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/ansible-doc: Cannot create symlink to ‘python.py’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/ansible-galaxy: Cannot create symlink to ‘python.py’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/ansible-inventory: Cannot create symlink to ‘python.py’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/ansible-playbook: Cannot create symlink to ‘python.py’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/ansible-pull: Cannot create symlink to ‘python.py’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/ansible-test: Cannot create symlink to ‘python.py’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/ansible-vault: Cannot create symlink to ‘python.py’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/importer.py: Cannot create symlink to ‘python.py’: No such file or directory
tar: ansible-2.9.7/test/lib/ansible_test/_data/injector/pytest: Cannot create symlink to ‘python.py’: No such file or directory
tar: Exiting with failure status due to previous errors
==> ERROR: A failure occurred in prepare().
```

**Pull request merged**